snapshot/testsuite/testsuite.go

```diff
@@ -435,6 +436,142 @@ func checkSnapshotterPrepareView(ctx context.Context, t *testing.T, snapshotter
 
 }
 
+// Deletion of files/folder of base layer in new layer, On Commit, those files should not be visible.
+func checkDeletedFilesInChildSnapshot(ctx context.Context, t *testing.T, snapshotter snapshot.Snapshotter, work string) {
```

Comment:
This test seems very similar to tests we have. Can you code up this test using the `fstest.Apply` method and `checkSnapshots`. Since this test only makes alterations to a single snapshot hierachy and tests the result, it is a good candidate for using the same pattern. In this case maybe you wanted to create an empty layer in between, not sure but I kind of see it like this...

```.go
	l1Init := fstest.Apply(
		fstest.CreateFile("/foo", []byte("foo\n"), 0777),
		fstest.CreateFile("/foobar", []byte("foobar\n"), 0777),
	)
	l2Init := fstest.Apply(
		fstest.RemoveAll("/foobar"),
	)
	l3Init := fstest.Apply() 

	if err := checkSnapshots(ctx, sn, work, l1Init, l2Init, l3Init); err != nil {
		t.Fatalf("Check snapshots failed: %+v", err)
	}
```

Reply:
I had tried to use ``fstest.*`` functions for this test.
The reason, I didn't used as there was no functions for ``Stat`` a file. 

In above example, does ``checkSnapshots()`` Verifies  if ``/foo`` exists with ``l3Init`` or just create a empty layer?.

Reply:
@kunalkushwaha checkSnapshots will verify that the layer is exactly the same as operations performed directly on the filesystem. It will indeed do a stat, but it will compare it against that operation having been performed in a flat directory. In the end these tests are trying to verify that these operations done in different layers are equivalent to the operations being performed on a flat directory.

Reply:
In the example I specified, l3Init just creates an empty layer. It is not necessary as `checkSnapshots` will always commit and create a view for checking. However if you wanted to test the change in a "middle" layer, it might make sense to add an empty layer on top.

Reply:
thanks for detailed explanation :). Suggested changes are already updated.